### ISSUE TITLE
Mimicking Keras Training API

### ISSUE THREAD

**Issue by abhmul** [abhmul/PyJet#5]
In order to be able to seamlessly swap pyjet models with keras models, pyjet needs to change to use the same API for training and interact the same way with training related keras objects (e.g. callbacks). In order to do so, the following needs to happen:

- [ ] Write a `compile` method for SLModel to pass in loss functions, optimizers, and metrics into
- [ ] Support multi-input, multi-loss, and multi-output in the same fashion as keras
- [ ] Make `fit_generator` create a training log for callbacks just like Keras
- [ ] Modify `fit_generator` to return a history object
- [ ] Replace save_state and load_state with save_weights and load_weights respectively
- [ ] Create save and load_model just like with keras
- [ ] Change impelentable method for the forward pass of the network to `call` not `forward` to use `forward` for some forward pass overhead
- [ ] Create non-generator equivalents (i.e. create Keras's `fit`, `evaluate`, and `predict` methods)

**Comment by abhmul** [comment]
#6 Could help with doing this properly

**Comment by abhmul** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Explore avenues for completely mimicking Keras API](https://trello.com/c/zzUP0wjr/8-explore-avenues-for-completely-mimicking-keras-api)
